core/src/main/scala/io/finch/internal/IdToEffect.scala

```diff
@@ -0,0 +1,10 @@
+package io.finch.internal
+
+import cats.effect.Effect
+import io.finch.ToEffect
+
+trait IdEffect {
```

Comment:
I lost our previous conversion on this, but I'm still not sure why we need these instances defined outside of `ToEffect` companion object?

Reply:
the rational is that comparing to put everything in one single big object, it's more extensible and modular by putting `ToEffect`'s instances in different files
the implementation of `ToEffect[E,E]` here is totally irrelevant to the implementation of `ToEffect[*Future, E]`(ideally TwtterFuture and ScalaFuture should be separated too but I think it's fine that they share some similarity)

Reply:
Extensible in a sense of for Finch users or for Finch developers? I'm trying to understand if there is any value in the latter.

Reply:
for finch developers, e.g. if finch want to natively support some new type other then Future, Id, it's easier to add new trait for `object ToEffect` to extends

user shouldn't care these traits so that's why I move them back to `internal` folder

Reply:
Sorry, I'm a little skeptical on what's easier: create a new trait and extend it in an object, or just put your stuff under the same object. I personally like the latter more b/c it's just less code and fewer indirection and fewer symbols. Perhaps, we can revisit it when we have 10-20 instances the grouping is the only vehicle to easier reasonability and discoverability.

Reply:
Does that means you're ok with a big object live in package.scala?
I personally really don't like putting irrelevant thing in single file, and keeping package.scala as concise as possible

Reply:
I'd prefer a separate file for this, `ToEffect.scala` but also curious to hear what other think.

Reply:
Agree with @vkostyukov 

Also, one thing to consider. If at some point we'd like to ship Monix support with finch, most likely it'll be separate `finch-monix` submodule anyway

Reply:
I don't think we need another modules to support monix , actually we probably don't need to do anything and monix should be supported already, since we have generic endpointmodule[E[_]], monix task is an Effect already or at least for sure it has Async implemention already

Reply:
@jcouyang It goes a bit further then just a `Mapper` support:
https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/package.scala#L11

It allows to enable simple syntax for all the `Endpoint.` methods

Reply:
I'm not talking about mapper, I'm talking about endpointmodule [F[_]]

The F is generic as Effect already isn't it?

Reply:
Shortly, `EndpointModule[F[_]]` is a container for all the aliases for `Endpoint.` methods such as `get`, `post`, `path` and so on. Otherwise user would neeed to type manually each time something like:
```scala
Endpoint.path[IO, String]("foo")

//equal to
import io.finch.catsEffect._
path("foo")
```

There are multiple ways to enable such syntax, one of them is to have the object extending `EndpointModule[AnyEffect]`. Details: https://github.com/finagle/finch/releases/tag/v0.25.0

Reply:
> I'd prefer a separate file for this, ToEffect.scala

Hmm🤔 A second thought, if move to separate file, object mapper can simpliy extend the trait and has the ability to convert none effect to effect, which is more composable and less implicit

And type alias for `F ~>E` can be eliminated too, user don't have to import anything from Finch to implement `F ~>E` for their own type then

Not sure if it makes sense to you, I'll refactor it to demonstrate what I meant anyway

Reply:
> object mapper can simpliy extend the trait and has the ability to convert none effect to effect

forget it, that theory didn't work out

> I'd prefer a separate file for this, ToEffect.scala

@vkostyukov just to be clear,  i dont want to change back and forth again
IIUC is this what you saying?
- combine IdEffect.scala and FutureToEffect.scala into `ToEffect.scala`

**or**
- move everything including the `trait ToEffect` from `package.scala` into `ToEffect.scala`